vault/diagnose/tls.go

```diff
@@ -4,24 +4,52 @@ import (
 	"crypto/tls"
 	"crypto/x509"
 	"fmt"
+	"io/ioutil"
 
 	"github.com/hashicorp/vault/internalshared/listenerutil"
+	"github.com/hashicorp/vault/sdk/helper/tlsutil"
 	"github.com/hashicorp/vault/vault"
 )
 
+const minVersionError = "'tls_min_version' value %q not supported, please specify one of [tls10,tls11,tls12,tls13]"
+const maxVersionError = "'tls_max_version' value %q not supported, please specify one of [tls10,tls11,tls12,tls13]"
+
 func ListenerChecks(listeners []listenerutil.Listener) error {
 	for _, listener := range listeners {
 		l := listener.Config
-		err := TLSFileChecks(l.TLSCertFile, l.TLSKeyFile)
+		var err error
```

Comment:
nitpick/style thing, but this could be declared closer to where it's actually used